app/src/org/commcare/provider/InstanceProvider.java

```diff
@@ -211,28 +205,15 @@ public Uri insert(@NonNull Uri uri, ContentValues initialValues) {
             Uri instanceUri = ContentUris.withAppendedId(InstanceProviderAPI.InstanceColumns.CONTENT_URI, rowId);
             notifyChangeSafe(getContext(), uri);
 
-            if (linkToSession) {
-                try {
-                    linkToSessionFormRecord(instanceUri);
-                } catch (IllegalStateException e) {
-                    throw e;
-                } catch (Exception e) {
-                    Logger.exception(e);
-                    throw new SQLException("Failed to insert row into " + uri);
-                }
-            } else {
-                // Forms with this flag are being loaded onto the phone
-                // manually and hence shouldn't be attached to the FormRecord
-                // in the current session
-                String xmlns = values.getAsString(InstanceProviderAPI.InstanceColumns.JR_FORM_ID);
-
-                SecretKey key;
-                key = CommCareApplication._().createNewSymmetricKey();
-                FormRecord r = new FormRecord(instanceUri.toString(), FormRecord.STATUS_UNINDEXED,
-                        xmlns, key.getEncoded(), null, new Date(0), mDbHelper.getAppId());
-                IStorageUtilityIndexed<FormRecord> storage =
-                        CommCareApplication._().getUserStorage(FormRecord.class);
-                storage.write(r);
+            switch (insertType) {
+                case SESSION_LINKED:
+                    finalizeSessionLinkedInsertion(instanceUri, uri);
+                    break;
+                case UNINDEXED_IMPORT:
+                    finalizeUnindexedInsertion(values, instanceUri);
+                    break;
+                case SANDBOX_MIGRATED:
+                    break;
```

Comment:
The fix is to add this migration branch that doesn't do anything to the form record.